### DISCUSSION TITLE
A verb to skip the first n lines

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#725] [Ideas]
Hi,
I have this kind of a zipped csv

```
Popolazione residente al 1° Gennaio 2021 per età sesso e stato civile - dati provvisori
Codice comune,Denominazione,Età,Totale Maschi,Totale Femmine
1001,Agliè,0,13,7
1001,Agliè,1,4,9
1001,Agliè,2,9,13
1001,Agliè,3,7,8
1001,Agliè,4,7,10
1001,Agliè,5,12,14
1001,Agliè,6,11,10
1001,Agliè,7,12,14
```

To read it I run

```
zcat input.zip| tail -n +2 |  mlr --csv cat
```

I need to skip the first row. I cannot use comments-in-data flags (there is no flag), and the miller tail verb can not be used to skip the first n rows.

What about to have a verb that forces miller to read the source starting from a certain line? Or to enable tail to do it, something like:

```
mlr --prepipe-gunzip --csv  tail -n +2 cat input.zip
```

Have a nice day

**Comment by sjackman** [comment]
I have a compressed Matrix Market file that looks like…
```
%%MatrixMarket matrix coordinate integer general
%metadata_json: {"software_version": "2022.0915.0", "format_version": 2}
19059 5151 8916925
7 1 1
…
```
I need to skip the lines that start with `%` and the first row `19059 5151 8916925` which indicates the number of rows, columns, and non-zero matrix entries (lines that follow). Easily enough done with `pizg -cd foo.mtx.gz | tail -n+4 | mlr --nidx …`, but a `--skip-lines=N` would be useful like `mlr --prepipe-gunzip --nidx --skip-comments-with=% --skip-lines=1`
